compiler/src/iree/compiler/Dialect/Flow/Transforms/DispatchLinalgOnTensors.cpp

```diff
@@ -802,6 +803,137 @@ static FailureOr<Flow::WorkloadBuilder> getWorkloadBuilder(OpBuilder &builder,
   return result;
 }
 
+/// Finds contiguous loop ranges from the given linalg.generic.
+static SmallVector<ReassociationIndices> getContiguousLoops(
+    linalg::GenericOp genericOp, SmallVectorImpl<unsigned> &loopDims) {
+  SmallVector<ReassociationIndices> contiguousLoops;
+  if (loopDims.size() < 2) return contiguousLoops;
+
+  auto isAllAffinemapsContiguous = [&](unsigned dim) {
+    return llvm::all_of(genericOp.getIndexingMapsArray(), [=](AffineMap map) {
+      return map.getDimPosition(dim) == dim;
```

Comment:
do we check somewhere that the maps are projected permutation? Otherwise you cannot assume that every result is an `AffineDimExpr`